sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveReferencesInAggregate.scala

```diff
@@ -112,9 +105,12 @@ object ResolveReferencesInAggregate extends SQLConfHelper
     assert(selectList.forall(_.resolved))
     if (conf.groupByAliases) {
       groupExprs.map { g =>
-        g.transformWithPruning(_.containsPattern(UNRESOLVED_ATTRIBUTE)) {
+        g.transformWithPruning(_.containsAnyPattern(UNRESOLVED_ATTRIBUTE,
+          LATERAL_COLUMN_ALIAS_REFERENCE)) {
```

Comment:
The pattern can guarantee that we have only `UnresolvedAttribute` and `LateralColumnAliasReference`, so you can write more generic code:
```scala
          case u : NamedExpression =>
            selectList.find(ne => conf.resolver(ne.name, u.name)).getOrElse(u)
```

Reply:
I'm not sure if this way will match cases that shouldn't be matched, because `_.containsAnyPattern(UNRESOLVED_ATTRIBUTE, LATERAL_COLUMN_ALIAS_REFERENCE` also return true when one plan children node contains `UnresolvedAttribute` or `LateralColumnAliasReference`.